modmesh/app/euler1d.py

```diff
@@ -530,7 +530,8 @@ def build_grid_figure(self):
 
         :return: FigureCanvas
         """
-        x = self.st.svr.coord[::2]
+        pt_mid = np.linspace(2,len(self.st.svr.coord) - 3,num=((len(self.st.svr.coord) - 3) // 2),dtype=int)
```

Comment:
This code appears many times, would you like to optimize it ?

Reply:
Yes, I think I can do it.